Add crd to kubernetes

rodrigofronza opened the pull request:
Add crd to kubernetes

Issue: #10

rodrigofronza committed:
Add crd to kubernetes

Issue: #10

Signed-off-by: rodrigofronza <rodrigofronza2@gmail.com>

rodrigofronza committed:
Add test-manifest function

Issue: #10

Signed-off-by: rodrigofronza <rodrigofronza2@gmail.com>

NunoFrRibeiro committed:
This is the way

I think

Issue: #10

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
After all this is the way

Issue: #10

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro commented:
Finally...
Calling dagger through the cli `dagger call test-cluster` runs a test cluster inside a container and installs the CRD's inside said cluster.
There could be a better and more safe way to do this, but for now it works.

TODO:
test positive and negative scenarios on the test-cluster function

@rodrigofronza, please test the command, the output should be:
```bash
❯  dagger call test-cluster
▶ connect 0.3s
▶ load module: . 16.6s
✔ parsing command line arguments 0.0s

✔ buildCi(
  ┆ source: Host.directory(path: "/home/repermon", noCache: true): Directory!
  ): BuildCi! 3.1s
▶ .testCluster: String! 40.1sv

customresourcedefinition.apiextensions.k8s.io/cicdmonitors.perf.my-org.io created
namespace/ai-monitor created
serviceaccount/cicd-operator-sa created
clusterrole.rbac.authorization.k8s.io/cicd-operator-role created
clusterrolebinding.rbac.authorization.k8s.io/cicd-operator-binding created
```

NunoFrRibeiro committed:
Update dagger version

Dagger engine/cli is updated to [v0.19.1](https://github.com/dagger/dagger/releases/tag/v0.19.1)

Issue: #10

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>

NunoFrRibeiro committed:
Update dagger module version

Issue: #10

Signed-off-by: NunoFrRibeiro <nunofrribeiro@hotmail.com>
